src/main/java/graphql/analysis/QueryTraversal.java

```diff
@@ -69,9 +70,8 @@ private QueryTraversal(GraphQLSchema schema,
         this.schema = assertNotNull(schema, "schema can't be null");
         this.variables = assertNotNull(variables, "variables can't be null");
         this.fragmentsByName = getOperationResult.fragmentsByName;
-        this.roots = getOperationResult.operationDefinition.getSelectionSet().getChildren();
+        this.roots = Collections.singletonList(getOperationResult.operationDefinition);
```

Comment:
I am a fan of static imports for readability

Reply:
changed